helm-charts/splunk-otel-collector/values.schema.json

```diff
@@ -289,7 +289,55 @@
           "type": "boolean"
         },
         "controlPlaneEnabled": {
-          "type": "boolean"
+          "type": "boolean",
+          "deprecated": true
+        },
+        "controlPlaneMetrics": {
+          "type": "object",
+          "properties:": {
+            "apiserver": {
+              "enabled": {
+                "type": "boolean"
+              }
+            },
+            "controllerManager": {
+              "enabled": {
+                "type": "boolean"
+              }
+            },
+            "coredns": {
+              "enabled": {
+                "type": "boolean"
+              }
+            },
+            "etcd": {
+              "enabled": {
+                "type": "boolean"
+              },
+              "type": "object",
```

Comment:
`additionalProperties` and secret related keys are missed.

Also it would be great if you can add validation secret.create=true or secret.name!=""

Reply:
Updated so all the properties are present. Added in some validations.